test/fluree/db/shacl/shacl_basic_test.clj

```diff
@@ -1291,3 +1291,125 @@
+               "rdf:type" ["ex:Kid"]
+               "ex:parent" [{"id" "ex:Bob"}
+                            {"id" "ex:Jane"}]}]
+             @(fluree/query valid-kid {"select" {"?s" ["*"]}
+                                       "where" [["?s" "id" "ex:ValidKid"]]})))
+      (is (util/exception? invalid-kid))
+      (is (= "SHACL PropertyShape exception - path [[1002 :predicate]] conformed to sh:qualifiedValueShape fewer than sh:qualifiedMinCount times."
+             (ex-message invalid-kid)))))
+  (testing "sh:qualifiedValueShapesDisjoint"
+    (let [conn   @(fluree/connect {:method :memory})
+          ledger @(fluree/create conn "shape-constaints" {:defaultContext [test-utils/default-str-context
+                                                                           {"ex" "http://example.com/"}]})
+          db0    (fluree/db ledger)
+
+          db1    @(fluree/stage db0 [{"id" "ex:Digit"
+                                      "ex:name" "Toe"}
+                                     {"id" "ex:HandShape"
+                                      "type" "sh:NodeShape"
+                                      "sh:targetClass" {"id" "ex:Hand"}
+                                      "sh:property" [{"sh:path" {"id" "ex:digit"}
+                                                      "sh:maxCount" 5}
+                                                     {"sh:path" {"id" "ex:digit"}
+                                                      "sh:qualifiedValueShape" {"sh:path" {"id" "ex:name"}
+                                                                                "sh:pattern" "Thumb"}
+                                                      "sh:qualifiedMinCount" 1
+                                                      "sh:qualifiedMaxCount" 1
+                                                      "sh:qualifiedValueShapesDisjoint" true}
+                                                     {"sh:path" {"id" "ex:digit"}
+                                                      "sh:qualifiedValueShape" {"sh:path" {"id" "ex:name"}
+                                                                                "sh:pattern" "Finger"}
+                                                      "sh:qualifiedMinCount" 4
+                                                      "sh:qualifiedMaxCount" 4
+                                                      "sh:qualifiedValueShapesDisjoint" true}]}])
+
+          valid-hand @(fluree/stage db1 [{"id" "ex:ValidHand"
+                                          "type" "ex:Hand"
+                                          "ex:digit" [{"ex:name" "Thumb"}
+                                                      {"ex:name" "Finger"}
+                                                      {"ex:name" "Finger"}
+                                                      {"ex:name" "Finger"}
+                                                      {"ex:name" "Finger"}]}])
+          invalid-hand @(fluree/stage db1 [{"id" "ex:ValidHand"
```

Comment:
Should the id be e.g. `{"id" "ex:InvalidHand"}` instead (just to clarify the intent; I see that it wouldn't affect the test either way)?